### ISSUE TITLE
cast_abs_to_unsigned generates non-compiling code when original code is in parens

### ISSUE THREAD

**Issue by shssoichiro** [rust-lang/rust-clippy#9185]
### Summary

When there is an expression using `.abs()` which requires parens, the suggestion generated by `cast_abs_to_unsigned` does not preserve those parens, resulting in code that does not compile.

### Reproducer

I tried this code:

```rust
let diff = (lossless_frames as i64 - dimensions.frames as i64).abs() as u32;
```

I expected to see this happen:

```
warning: casting the result of `i64::abs()` to u32
   --> src/output/video.rs:174:20
    |
174 |         let diff = (lossless_frames as i64 - dimensions.frames as i64).abs() as u32;
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace with: `(lossless_frames as i64 - dimensions.frames as i64).unsigned_abs()`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#cast_abs_to_unsigned
```

Instead, this happened:

```
warning: casting the result of `i64::abs()` to u32
   --> src/output/video.rs:174:20
    |
174 |         let diff = (lossless_frames as i64 - dimensions.frames as i64).abs() as u32;
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace with: `lossless_frames as i64 - dimensions.frames as i64.unsigned_abs()`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#cast_abs_to_unsigned
```

(note the lack of parens in the suggestion)

### Version

```text
rustc 1.64.0-nightly (23e21bdd2 2022-07-15)
binary: rustc
commit-hash: 23e21bdd25026e2839ebe946c2a937c1904887d2
commit-date: 2022-07-15
host: x86_64-unknown-linux-gnu
release: 1.64.0-nightly
LLVM version: 14.0.6
```

### Additional Labels

@rustbot label +I-suggestion-causes-error
